src/classes/CRLP_Batch_Base_NonSkew.cls

```diff
@@ -323,4 +322,84 @@ public abstract class CRLP_Batch_Base_NonSkew extends CRLP_Batch_Base {
+        Customizable_Rollup_Settings__c customizableRollupSettings = UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
+        if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
+                && customizableRollupSettings.AccountHardCreditNonSkew_Incremental__c == true) {
+            return true;
+        } else if (jobType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit
+                && customizableRollupSettings.ContactHardCreditNonSkew_Incremental__c == true) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @description Produces the where clause that should be used for the inner join when in incremental mode
+     * @return The WHERE clause, as a string, that should be used to filter the detail records
+     */
+    private String getIncrementalModeWhereClause() {
+        String whereClauseDateField = Test.isRunningTest() ? 'CloseDate' : 'SystemModstamp'; // Set to CloseDate
+        // or another Date field for testing.
+        // Will be set to SystemModstamp for prod.
+
+        String whereClause = 'Id IN (' +
+                'SELECT ' + keyField + ' ' +
+                'FROM ' + detailObjectType + ' ALLROWS ' +
+                'WHERE ';
+
+        if (maxDaysAgo != null) {
+            whereClause += whereClauseDateField + ' >= LAST_N_DAYS:' + maxDaysAgo;
+        }
+        if (maxYearsAgo != null) {
+            if (maxDaysAgo != null) {
+                whereClause += ' OR ';
+            }
+            whereClause += whereClauseDateField + ' >= LAST_N_YEARS:' + maxYearsAgo;
+        }
+        return whereClause += ')';
+    }
+
+    /**
+     * @description Produces the where clause that should be used for the inner join when NOT in incremental mode
+     * @return The WHERE clause, as a string, that should be used to filter the detail records
+     */
+    private String getStandardWhereClause() {
+        String standardWhereClause = null; //will return null if this method is called from an unsupported rollup type
+        if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
+                || jobType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit) {
+            standardWhereClause = 'Id IN (' +
+                    'SELECT ' + keyField + ' ' +
+                    'FROM ' + detailObjectType + ' ALLROWS' +
+                    ')';
+        }
+        return standardWhereClause;
+    }
+
+    /**
+     * @description Produces the where clause that should be used for the inner join
+     * @return The WHERE clause, as a string, that should be used to filter the detail records
+     */
+    @TestVisible private String getWhereClause() {
+        Boolean incrementalModeActivatedForThisJobType = isIncrementalMode(jobType);
+        Boolean hasActiveIncrementalModeRollups = (maxDaysAgo != null || maxYearsAgo != null);
```

Comment:
I think this is more subtle in what we can support for this. For example, recurring donations do not have a native rollup that uses Days or Years, so in that case both of these would be null. However, that doesn't necessarily mean that an incremental rollup wouldn't be beneficial. I recommend that if both fields are null, the range default to 1 year (365 days). This way something like Recurring Donation rollups would see improved performance with its rollups.

NOTE: Since RD's aren't covered in this iteration, how about if the code make include a TODO comment to reflect the fact that we want to modify this to support rollups where there is no time period.